devdoc/waiter_queue_requirements.md

```diff
@@ -0,0 +1,103 @@
+# `waiter_queue` requirements
+
+## Overview
+
+`waiter_queue` is a module that allows consumers to park calls and for producers to signal that the parked calls be executed. A "parked call" consists of a callback function and a context. The consumer provides the callback function and context when it parks a call. The consumer expects the parked call's callback function to be executed with the provided context when the producer signals.
+
+`waiter_queue` has two parts: `waiter_queue_ll` that contains the logic for parking and unblocking calls and `waiter_queue_hl` that contains logic for making `waiter_queue` thread-safe.
```

Comment:
>`waiter_queue_ll` that contains the logic for parking and unblocking calls and `waiter_queue_hl` [](http://example.com/codeflow?start=30&length=96)

poetry: isn't this `waiter_queue` and `threadsafe_waiter_queue` or something? /consider